easybuild/framework/easyblock.py

```diff
@@ -1575,7 +1662,68 @@ def sanity_check_step(self, custom_paths=None, custom_commands=None, extension=F
             raise EasyBuildError("Incorrect format for sanity_check_paths (should (only) have %s keys, "
                                  "values should be lists (at least one non-empty)).", ','.join(req_keys))
 
+        commands = self.cfg['sanity_check_commands']
+        if not commands:
+            if custom_commands:
+                commands = custom_commands
+                self.log.info("Using customised sanity check commands: %s" % commands)
+            else:
+                commands = []
+                self.log.info("Using specified sanity check commands: %s" % commands)
+
+        for i, command in enumerate(commands):
+            # set command to default. This allows for config files with
+            # non-tuple commands
+            if not isinstance(command, tuple):
+                self.log.debug("Setting sanity check command to default")
+                command = (None, None)
+
+            # Build substition dictionary
+            check_cmd = {
+                'name': self.name.lower(),
+                'options': '-h',
+            }
+            if command[0] is not None:
+                check_cmd['name'] = command[0]
+            if command[1] is not None:
+                check_cmd['options'] = command[1]
+
+            commands[i] = "%(name)s %(options)s" % check_cmd
+
+        return paths, path_keys_and_check, commands
+
+    def _sanity_check_step_dry_run(self, custom_paths=None, custom_commands=None, **_):
+        """Dry run version of sanity_check_step method."""
+        paths, path_keys_and_check, commands = self._sanity_check_step_common(custom_paths, custom_commands)
+
+        for key in path_keys_and_check:
+            dry_run_msg("Sanity check paths - %s" % key, silent=self.silent)
+            if paths[key]:
+                for path in sorted(paths[key]):
+                    dry_run_msg("  * %s" % str(path), silent=self.silent)
+            else:
+                dry_run_msg("  (none)", silent=self.silent)
+
+        dry_run_msg("Sanity check commands", silent=self.silent)
+        if commands:
+            for command in sorted(commands):
+                dry_run_msg("  * %s" % str(command), silent=self.silent)
+        else:
+            dry_run_msg("  (none)", silent=self.silent)
+
+    def _sanity_check_step(self, custom_paths=None, custom_commands=None, extension=False):
+        """Real version of sanity_check_step method."""
+        paths, path_keys_and_check, commands = self._sanity_check_step_common(custom_paths, custom_commands)
+
+        # check sanity check paths
         for key, check_fn in path_keys_and_check.items():
+            if key == 'dirs':
+                typ = "(non-empty) directory"
```

Comment:
to avoid duplicate code? ;)

will check, this could make the output of the dry run sanity_check_step clearer, indeed